### DISCUSSION TITLE
'youtube-dl' is not recognized as an internal or external command, operable program or batch file.

### DISCUSSION THREAD

**Discussion by asdigisoftech** [norkunas/youtube-dl-php#187] [Q&A] [answered]
I'm getting this error when trying to download youtube video.

This is the code. I have set path of **youtube-dl** in environment variable.

`$collection = $yt
    ->setBinPath('youtube-dl')
    ->download(
    Options::create()
        ->downloadPath('downloads')
        ->url('https://www.youtube.com/watch?v=oDAw7vW7H0c')
);`

When I'm printing result buffer then I'm getting the error "'youtube-dl' is not recognized as an internal or ......."

`$process->run(function (string $type, string $buffer) use (&$currentVideo, &$parsedData, &$progressTarget): void {
            ($this->debug)($type, $buffer);
            echo "<pre>"; print_r($buffer); die();`

Any help would be appreciable.

Thank you

**Comment by norkunas** [comment, marked as answer]
Hello, sorry for late response.

Your `setBinPath` should set the absolute path to the binary.
